CI/unit_tests/graph_modules/test_molecular_graph.py

```diff
@@ -0,0 +1,124 @@
+Summary
+-------
+Module to test the molecular graph module.
+"""
+from pathlib import Path
+
+import numpy as np
+import tensorflow as tf
+
+from mdsuite.graph_modules.molecular_graph import MolecularGraph
+
+
+class MockExperiment:
+    """
+    Experiment class for a unit test
+    """
+
+    database_path = Path("./")
+
+
+class TestMolecularGraph:
+    """
+    Class to test the molecular graph module.
+    """
+
+    graph_class = MolecularGraph(experiment=MockExperiment())
+
+    def test_apply_system_cutoff(self):
+        """
+        Test the apply_system_cutoff method.
+
+        Returns
+        -------
+        Checks whether or not the cutoff has been enforced.
+        """
+        zeros = np.array([0, 0, 0, 0, 0])
+        cutoff_data = [
+            [0, 6.8, 9, 11, 0],
+            [12, 0.0, 4, 3.27, 1],
+            [200, 38, 0, 1, 2.11],
+            [5, 5, 5.8765342, 0, 5],
+            [1, 7, 3, 4.8762511, 0],
+        ]
+        cutoff_tensor = tf.constant(cutoff_data, dtype=tf.float32)
+
+        # Middle range cutoff
+        target = tf.constant(
+            [
+                [0, 0, 0, 0, 1],
+                [0, 0, 1, 1, 1],
+                [0, 0, 0, 1, 1],
+                [0, 0, 0, 0, 0],
+                [1, 0, 1, 1, 0],
+            ]
+        )
+        mask = self.graph_class._apply_system_cutoff(cutoff_tensor, cutoff=5)
+        np.testing.assert_array_equal(np.diagonal(mask), zeros)
+        np.testing.assert_array_equal(mask, target)
+
+        # All zeros
+        target = tf.zeros((5, 5))
+        mask = self.graph_class._apply_system_cutoff(cutoff_tensor, cutoff=0)
+        np.testing.assert_array_equal(np.diagonal(mask), zeros)
+        np.testing.assert_array_equal(mask, target)
+
+        # All ones
+        target = tf.ones((5, 5)) - tf.eye(5)
+        mask = self.graph_class._apply_system_cutoff(cutoff_tensor, cutoff=200.01)
+        np.testing.assert_array_equal(np.diagonal(mask), zeros)
+        np.testing.assert_array_equal(mask, target)
+
+    def test_build_smiles_graph(self):
+        """
+        Test the build_smiles_graph method.
+
+        Returns
+        -------
+        This test checks that the SMILES graphs built by the module return the correct
+        molecule information for several scenarios.
+        """
+        emim = "CCN1C=C[N+](+C1)C"
+        bmim = "CCCCN1C=C[N+](+C1)C"
+        pf = "F[P-](F)(F)(F)(F)F"  # PF6
+        h2o = "[H]O[H]"
+        bergenin = "OC[C@@H](O1)[C@@H](O)[C@H](O)[C@@H]2[C@@H]1c3c(O)c(OC)c(O)cc3C(=O)O2"
+        nacl = "[Na+].[Cl-]"
+
+        data = {
+            emim: {"species": {"C": 6, "N": 2, "H": 12}, "nodes": 20},
+            bmim: {"species": {"C": 8, "N": 2, "H": 16}, "nodes": 26},
+            pf: {"species": {"P": 1, "F": 6}, "nodes": 7},
+            h2o: {"species": {"H": 2, "O": 1}, "nodes": 3},
+            bergenin: {"species": {"C": 14, "H": 16, "O": 9}, "nodes": 39},
+            nacl: {"species": {"Na": 1, "Cl": 1}, "nodes": 2},
```

Comment:
In the test or in general? If it is in general going to be a data class then I will need to update the example to teach the users how to create a data class. Or we provide a templated class, which while typing it sounds quite nice, e.g some molecule class which takes all of this information.

Reply:
I would make it a dataclass in general but keep species a dict to make interfacing easier. Something like:

```py
from mlsuite import Molecule # or similar

molecule = Molecule(species={"P": 1, "F": 6}, nodes=7)
```

Reply:
yep and then I guess have the option of passing a smiles or a dict here. In the case of what we are referring to here this is actually the output of a different module so these are just for but the user input stuff I will make a data class. This stuff here is just for testing, the user doesn't give it this information.

Reply:
@PythonFZ and @christophlohrmann just to clarify here. I can make this a data class but these dicts here are 100% only restricted to this test. They have nothing to do with user interfacing or anything and purely here for as a convenience during solely this test. The user-interfaced molecule stuff will be made a data-class but do both of you see a point in making this also a data class? and just to stress, it would be a data-class made in this test module for this one test, not a part of the mdsuite package.

Reply:
For better readbility I think it would still be worth changing it.

Reply:
aded in ec1f25d